tests/test_move.py

```diff
@@ -151,6 +151,14 @@ def test_move_file_read_only_mem_dest(self):
                 dst_ro.exists("target.txt"), "file should not have been copied over"
             )
 
+    def test_overwrite(self):
+        with open_fs("temp://") as src, open_fs("temp://") as dst:
```

Comment:
Might be worth adding a comment here that this is specifically using TempFS instead of MemFS (as the other tests do) in order to test the syspath-optimised code?

Reply:
Good suggestion, added this!